docs/painless/painless-contexts/painless-ingest-processor-context.asciidoc

```diff
@@ -38,4 +38,145 @@ void::
+           dts, DateTimeFormatter.ISO_OFFSET_DATE_TIME);
+<15> ctx.datetime = dt.getLong(ChronoField.INSTANT_SECONDS)*1000L;
+----
+<1> Creates a `split` <<painless-functions, function>> to split a
+    <<string-type, `String`>> type value using a <<primitive-types, `char`>>
+    type value as the delimiter. This is useful for handling the necessity of
+    pulling out the individual pieces of the date and time `Strings` from the
+    original seat data since .
+<2> The first pass through each `char` in the `String` collects how many new
+    `Strings` the original is split into.
+<3> Returns the original `String` if there are no instances of the delimiting
+    `char`.
+<4> Creates an <<array-type, array type>> value to collect the split `Strings`
+    into based on the number of `char` delimiters found in the first pass.
+<5> The second pass through each `char` in the `String` collects each split
+    substring into an array type value of `Strings`.
+<6> Collects the last substring into the array type value of `Strings`.
+<7> Uses the `split` function to separate the date `String` from the seat data
+    into year, month, and day `Strings`. Note the use of a `String` type value
+    to `char` type value <<string-character-casting, cast>> as part of the
+    second argument since character literals do not exist.
+<8> Appends the <<string-literals, string literal>> `"0"` value to a single
+    digit month since the format of the seat data allows for this case.
+<9> Appends the <<string-literals, string literal>> `"0"` value to a single
+    digit day since the format of the seat data allows for this case.
+<10> Sets the <<primitive-types, `boolean type`>>
+     <<painless-variables, variable>> to `true` if the time `String` is a time
+     in the afternoon or evening.
+<11> Uses the `split` function to separate the time `String` from the seat data
+     into hours and minutes `Strings`. Note the use of the `substring` method to
+     remove the `AM` or `PM` portion of the time `String`. Also note the use of
+     a `String` type value to `char` type value
+     <<string-character-casting, cast>> as part of the second argument since
+     character literals do not exist.
+<12> If the time `String` is an afternoon or evening value adds the
+     <<integer-literals, integer literal>> `12` to the existing hours to move to
+     a 24-hour based time.
+<13> Builds a new time `String` that is parsable using existing API methods.
+<14> Creates a `ZonedDateTime` <<reference-types, reference type>> value by using
+     the API method `parse` to parse the new time `String`.
+<15> Sets the datetime field `datetime` to the number of milliseconds retrieved
+     from the API method `getLong`. Note the use of the `ctx` ingest processor
+     context variable to set the field `datetime`. Manipulate each document's
+     fields with the `ctx` variable as each document is indexed.
+
+Use the following curl command to create the ingest script processor:
+
```

Comment:
As above--conflating console vs cURL. I generally phrase this as, "Submit the following request to xyz:"

Reply:
Fixed.